lib/gitlab_api/create_hook.rb

```diff
@@ -0,0 +1,27 @@
+# frozen_string_literal: true
+
+module GitlabApi
+  class CreateHook < GitlabApi::Base
```

Comment:
What is the reason for using inheritance here?

Reply:
Well, the reason is so I can use the post method created on Base, without having to repeat code.

There is supposed to be an service for an controller to communicate with, so when you llink an story with an branch the service would fire up the CreateHook class in order to automatically create the necessary webhooks. I could swear I had created that service, I guess I got caught up into other stuff, might aswell do it now.

Reply:
Can we use composition instead of inheritance? Having "base classes" is usually a sign that we shouldn't be using inheritance because it won't pass the ["is a"](https://en.wikipedia.org/wiki/Is-a) test. You can do it by having an variable called `gitlab_api` in each hook and use what you need from it the hook.